### ISSUE TITLE
[Suggestion] Attempt to “emulate” installed battery, if possible

### ISSUE THREAD

**Issue by lunathir** [nukeykt/Nuked-SC55#23]
I figure that, if it’s somehow possible to make the emulator see there’s a battery installed, as well as saving and loading the emulated SC-55’s RAM on program exit and program load respectively, maybe that would get rid of the whole “you need to reset the MIDI module” thing as is the case right now, where it always boots into a buggy state if it hasn’t loaded yet, plus it might get rid of the “no battery” warning that pops up when it loads.

**Comment by johnnovak** [comment]
Starting with a known-good-state restored by default at startup seems sensible. That would the the state of the SC-55 after a GS reset SysEx.

Emulating the janky authentic behaviour could be then an option I guess, but not many people would desire such randomness.

**Comment by lunathir** [comment]
The "janky authentic behavior" actually stems from the battery being dead/nonexistent, which clears RAM if it's unplugged, which currently the emulator acts like. I figured the emulator saving a 32k file when it quits and reading it when it loads would make the SC55 act like it's plugged in/has a battery.

**Comment by johnnovak** [comment]
> The "janky authentic behavior" actually stems from the battery being dead/nonexistent, which clears RAM if it's unplugged, which currently the emulator acts like. I figured the emulator saving a 32k file when it quits and reading it when it loads would make the SC55 act like it's plugged in/has a battery.

Yeah, although that could still result in surprising results. Hence I was effectively suggesting to save the state after turning on followed by a GS reset into a 32k file, then just always use that.

But I guess preserving the RAM could be an option too.

**Comment by lunathir** [comment]
I get that, and I get it would probably be what some others would do. Not my particular sort of preference as I'd just like it to work like it does with a battery, but I can understand some other people would want it to reset by default to just make sure everything works as it should.

**Comment by johnnovak** [comment]
Yeah just make it configurable.

**Comment by HeyItsLollie** [comment]
Anything that would save me from having to perform an INIT ALL every time I boot Nuked-SC55 (Q >> Hold Y+U then Q >> W), just to fix the SC-55mkII starting with its key transposition in an "off by one" state.
